File: Java/src/com/dbc/algorithm/Array/NumberOfEquivalentDominoPairs.java
package com.dbc.algorithm.Array;

public class NumberOfEquivalentDominoPairs {
    public static int numEquivDominoPairs(int[][] dominoes) {
        int[] remain = new int[100];
        int result = 0;
        for (int[] dominoe : dominoes) {
            int val = dominoe[0] >= dominoe[1] ? dominoe[0] * 10 + dominoe[1] : dominoe[1] * 10 + dominoe[0];
            result += remain[val];
            remain[val]++;
        }
        return result;
    }
}

File: Python/array/number_of_equivalent_domino_pairs.py
def numEquivDominoPairs(dominoes: list) -> int:
    remain = [0] * 100
    ret = 0
    for x, y in dominoes:
        val = (x * 10 + y if x <= y else y * 10 + x)
        ret += remain[val]
        remain[val] += 1
        
    return ret


if __name__ == "__main__":
    print(numEquivDominoPairs(dominoes=[[1,2],[1,2],[1,2],[1,1],[1,2],[2,2]]))
